### PR TITLE
feat: remove dockerfile-based builds

### PR DESCRIPTION
**Pull request by linostar**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

---

Remove dockerfile-based builds.

*Picture of a cool ROCK:*
image

### CODE REVIEW

**Comment by linostar** [comment]
@cjdcordeiro Ready for review.

File: .github/ISSUE_TEMPLATE/--onboarding.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ body:
     id: type
     attributes:
       label: Image type
-      description: Is this a ROCK or a Dockerfile-based image?
+      description: description of the rock image
```
Comment (cjdcordeiro) [inline .github/ISSUE_TEMPLATE/--onboarding.yaml:51]: this whole field can actually be removed no? if we don't support docker images anymore, there's no option -> it must be a rock

File: .github/workflows/Image.yaml [hunk 2]
```diff
@@ -149,65 +149,25 @@ jobs:
...
-            fi
-          else
-            grep 'FROM' Dockerfile | tail -1 | grep ubuntu || \
-              (echo "ERROR: the image '${rock_name}' must be based on Ubuntu!" \
-                && exit 1)
+            echo "ERROR: the ROCK's name '${rock_name}' must match the OCI folder name!"
+            exit 1
           fi
 
       # If this is a ROCK...
```
Comment (cjdcordeiro) [inline .github/workflows/Image.yaml:159]:
```suggestion
```

File: .github/workflows/Image.yaml [hunk 3]
```diff
@@ -223,7 +183,7 @@ jobs:
     with:
       oci-image-name: "${{ matrix.name }}_${{ matrix.commit }}_${{ matrix.revision }}"
       oci-image-path: "oci/${{ matrix.name }}"
-      is-a-rock: ${{ matrix.dockerfile-build == '' && true || false }}
+      is-a-rock: true
```
Comment (cjdcordeiro) [inline .github/workflows/Image.yaml:186]: well, if this is always true, we might as well get rid of it and assume it is true

File: src/uploads/infer_image_track.py [hunk 4]
```diff
@@ -25,51 +23,33 @@ def get_release_from_codename(codename: str) -> str:
...
 
-    rock_base = (
-        rockcraft_yaml["base"]
-        if rockcraft_yaml["base"] != "bare"
-        else rockcraft_yaml["build-base"]
+try:
+    base_release = float(rock_base.replace(":", "@").split("@")[-1])
+except ValueError:
+    logging.warning(
+        f"Could not infer ROCK's base release from {rock_base}. Trying with codename."
```
Comment (cjdcordeiro) [inline src/uploads/infer_image_track.py:46]:
```suggestion
        f"Could not infer rock's base release from {rock_base}. Trying with codename."
```

File: src/uploads/infer_image_track.py [hunk 5]
```diff
@@ -25,51 +23,33 @@ def get_release_from_codename(codename: str) -> str:
...
-        )
-        base_release = float(
-            get_release_from_codename(rock_base.replace(":", "@").split("@")[-1])
-        )
-
-    version = rockcraft_yaml["version"]
+version = rockcraft_yaml["version"]
 
 track = f"{version}-{base_release}"
 print(f"ROCK track: {track}")
```
Comment (cjdcordeiro) [inline src/uploads/infer_image_track.py:55]:
```suggestion
print(f"Rock track: {track}")
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
Thanks. It's getting there. It's a bit of a tedious task, but the following references still need updating:
 - IMAGE_MAINTAINER_AGREEMENT.md
 - README.md (still has text referring to "Dockerfile-based" images)
 - same for --onboarding.yml
 - Tests.yaml still has `is-a-rock` and at least one step that is conditional to it being a rock

**Comment by linostar** [comment]
Replaced by [PR #105](https://github.com/canonical/oci-factory/pull/105).

**Pull request closed without merging**